compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2082,7 +2082,7 @@ class Typer extends Namer
   }
 
   private def checkSpliceOutsideQuote(tree: untpd.Tree)(implicit ctx: Context): Unit = {
-    if (level == 0 && !ctx.owner.isInlineMethod)
+    if (level == 0 && !ctx.owner.ownersIterator.exists(_.is(Inline)))
```

Comment:
What if the following code?

```Scala
inline def foo() = {
  class A {
    def bar = $(...)
  }
}
```

Reply:
It will fall back to error message of the macro body checking .

Reply:
Those checks are performed by the `Splicer`